tools/reviewer/job/sync/DiffConverter.java

```diff
@@ -0,0 +1,205 @@
+          .setOwner(githubPr.getRepo().getOwner())
+          .setAssociatedReviewerDiff(diff.getId());
+      pullRequests.add(pullRequest.build());
+    }
+    return pullRequests;
+  }
+
+  private List<ReviewComment> getReviewCommentsByRepoName(Diff diff, GithubPr githubPr) {
+    List<ReviewComment> result = new ArrayList<>();
+    List<Thread> codeThreads =
+        diff.getCodeThreadList()
+            .stream()
+            .filter(thread -> thread.getRepoId().equals(githubPr.getRepo().getRepo()))
+            .collect(Collectors.toList());
+
+    for (Thread thread : codeThreads) {
+      for (Comment comment : thread.getCommentList()) {
+        ReviewComment.Builder githubComment = ReviewComment.newBuilder();
+        githubComment
+            .setPath(thread.getFile().getFilename())
+            .setId(comment.getGithubCommentId())
+            .setPosition(getReviewCommentPosition(thread, githubPr))
+            .setCommitId(thread.getFile().getCommitId())
+            .setUser(User.newBuilder().setEmail(comment.getCreatedBy()).build())
+            .setBody(comment.getContent())
+            .setCreatedAt(String.valueOf(Instant.ofEpochMilli(comment.getTimestamp())))
+            .setReviewerThreadId(thread.getId())
+            .build();
+        result.add(githubComment.build());
+      }
+    }
+    return result;
+  }
+
+  // TODO: Check if the state is correct
+  private String getPullRequestState(Diff.Status status) {
+    if (status.equals(Diff.Status.ACCEPTED)) {
+      return "close";
+    } else {
+      return "open";
+    }
+  }
+
+  private CommitPointer getCommitPointer(
+      GithubPr githubPr, CommitPointerType commitPointerType, String headBranch)
+      throws IOException {
+
+    if (githubPr.getNumber() == 0) {
+      CommitPointer.Builder commitPointer = CommitPointer.newBuilder();
+      commitPointer.setRef(
+          commitPointerType.equals(CommitPointerType.BASE) ? "master" : headBranch);
+      return commitPointer.build();
+    } else {
+      PullRequest existingPullRequest =
+          githubClient
+              .getPullRequest(
+                  GithubProtos.PullRequestRequest.newBuilder()
+                      .setOwner(githubPr.getRepo().getOwner())
+                      .setRepo(githubPr.getRepo().getRepo())
+                      .setNumber(githubPr.getNumber())
+                      .build())
+              .getPullRequest();
+
+      if (commitPointerType.equals(CommitPointerType.BASE)) {
+        return existingPullRequest.getBase();
+      } else {
+        return existingPullRequest.getHead();
+      }
+    }
+  }
+
+  private int getReviewCommentPosition(Thread thread, GithubPr githubPr) {
+    LineNumberConverter.Side commentSide =
+        thread.getFile().getCommitId().equals(thread.getCommitId())
+            ? LineNumberConverter.Side.RIGHT
+            : LineNumberConverter.Side.LEFT;
+    String patch = "";
+    try {
+      patch =
```

Comment:
We're calling GitHub here to get the comment position, right?
How about we save the comment position just like we save the `github_comment_id`?
We'd like to avoid making O(#comments) calls to GitHub.

Reply:
Added new field to `Diff.Thread` message: `int32 github_comment_position`.
But I left this method too. Because a new comment (created by Reviewer) doesn't have GitHub comment position. It will be setted after the first sync. We check:
```
 if (thread.getGithubCommentPosition() == 0) {
send request to GitHub
} else {
return thread.gitHubCommentPosition
}
```

Reply:
Ok. Is `0` a valid comment position? For example, a comment on the first line.

Reply:
That's fine.